core/src/Streamly/Internal/Data/Stream/Chunked.hs

```diff
@@ -987,8 +997,11 @@ runArrayFoldManyD
                         return $ D.Skip $ ParseChunksBuf [x] s [] ps
                     PRD.IDone pb ->
                         let next = ParseChunksInit [x] s
-                         in return $ D.Skip $ ParseChunksYield pb next
-                    PRD.IError err -> throwM $ ParseError err
+                         in return $ D.Skip $ ParseChunksYield (Right pb) next
+                    PRD.IError err ->
+                        let next = ParseChunksInit [x] s
```

Comment:
We cannot continue even after error, this is wrong. Please see the parseMany implementation in StreamD for reference. We need to go to the ParseChunksInitLeftover state with empty buffer to stop the stream after error. This needs to be fixed everywhere in this function.